src/libraries/System.Net.Quic/tests/FunctionalTests/QuicStreamTests.cs

```diff
@@ -434,6 +435,43 @@ public async Task Read_StreamAborted_Throws()
                 Assert.Equal(ExpectedErrorCode, ex.ErrorCode);
             }).WaitAsync(TimeSpan.FromSeconds(15));
         }
+
+        [Fact]
+        public async Task WriteCanceled_NextWriteThrows()
+        {
+            long expectedErrorCode = 1234;
+
+            await RunClientServer(
+                clientFunction: async connection =>
+                {
+                    await using QuicStream stream = connection.OpenUnidirectionalStream();
+
+                    CancellationTokenSource cts = new CancellationTokenSource();
+                    var task = stream.WriteAsync(new byte[1024 * 1024], cts.Token);
+                    cts.Cancel();
```

Comment:
This way of testing cancellation is not reliable. Sometimes it seems that task finishes before cancellation occurs. Does anyone has an idea how to make sure write is not over before cancellation? I don't like an idea of pre-cancelling before calling to write...

Reply:
I can only think of ways which touch `MsQuicStream` code so not help, sorry.
But I think that adding extra test for pre-cancellation is actually a good idea. I'm looking at the code and it looks like we will send the data even if the token has been cancelled beforehand. Is that expected behavior? Should we do some best-effort token check before we call into msquic? Or did I just overlook something :blush:

Reply:
Sending 64M instead of 1M helped, at least I didn't get any failures on my setup while running it ~20 times. If someone believes it is not reliable enough, the only option I see here is to remove this test completely. I've added a second one that checks pre-cancelled token.

Reply:
Rewrote to an infinite loop until canceled, as per @geoffkizer suggestion

Reply:
Could that infinite loop be somehow finite, but still big enough for the test? I already see hanging tests in CI :)